_posts/2024-04-29-wait-you-guys-are-writing-unit-tests/wait-you-guys-are-writing-unit-tests.Rmd

```diff
@@ -0,0 +1,139 @@
+---
+title: "Wait, you guys are writing unit tests?"
+description: |
+  A reminder.
+author:
+  - name: Chloe Pou-Prom
+    url: {}
+date: 2024-04-29
+output:
+  distill::distill_article:
+    self_contained: false
+categories:
+    - post-miscellaneous
+draft: true
+preview: unit_tests.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+![](unit_tests.png)
+
+Confession: I rarely write unit tests. 😅 [^1]
+
+[^1]: Yes, I know. They're important and they help find critical failures. Just enter "Are unit tests worth it?" in your favorite search engine, and you'll end up with a mountain ⛰️ of blog posts, StackOverflow replies, Reddit comments, heated discussions, etc. that tell you: *yes, unit tests are worth it*.
+
+That being said, I did recently run into an issue that was identified by a unit test!!
+
+## What happened?
+
+**The situation**: We recently updated Data Virtuality (DV).
+
+> [Data Virtuality (DV)](https://datavirtuality.com/) is our data integration platform which consolidates data from various hospital source systems. It offers many advantages:
+>
+> -   DV has a layer of abstraction between source systems and data consumers. This means that, as a data scientist, I can access real-time data from one view.
+>
+> -   Any changes to source systems are dealt with at the virtualization layer and don't affect the user view.
+>
+> -   Query results can be reused in different places to avoid pulling from the source system multiple times.
+>
+> -   Additionally, if one of the source system fails, data from the previous update are cached.
+
+Many of our deployed applications rely on DV. It's *a vital production tool for our team*.
+
+### The Big Update
+
+It was time to update the DV version.
+
+**Big updates are always scary** so we had started preparing for this a while ago. We had a *protocol* in place. We were headed into this upgrade, ready to roll back to an earlier version if ever anything went wrong.
+
+The upgrade was happening after hours to limit the impact on end-users. I wasn't directly involved in the upgrade, but I was on standby with my phone set to max volume in case I received a panicked call in the middle of the night.
+
+The Big Day arrived and...
+
+...the upgrade went well!
```

Comment:
for the purposes of this post anyway :joy:

Reply:
🤭 😬